packages/ui5-task-pwa-enabler/lib/pwaEnabler.js

```diff
@@ -0,0 +1,286 @@
+    offlineCopy: "Offline-Copy",
+    preCache: "Cache-First",
+    advancedCache: "Advanced-Caching"
+};
+
+let default_manifest = {
+    "short_name": "MyApp",
+    "name": "My App Title",
+    "description": "Test App for PWA",
+    "icons": [
+        {
+            "src": "/images/icons/192x192.png",
+            "type": "image/png",
+            "sizes": "192x192"
+        },
+        {
+            "src": "/images/icons/512x512.png",
+            "type": "image/png",
+            "sizes": "512x512"
+        }
+    ],
+    "start_url": "/index.html",
+    "scope": "/",
+    "background_color": "#EFF4F9",
+    "theme_color": "#3F5161",
+    "display": "standalone"
+};
+
+/**
+ * UI5 PWA enabler
+ *
+ * @param {Object} parameters Parameters
+ * @param {module:@ui5/fs.DuplexCollection} parameters.workspace DuplexCollection to read and write files
+ * @param {module:@ui5/fs.AbstractReader} parameters.dependencies Reader or Collection to read dependency files
+ * @param {Object} parameters.options Options
+ * @param {string} parameters.options.projectName Project name
+ * @param {string} [parameters.options.projectNamespace] Project namespace if available
+ * @param {Object} [parameters.options.configuration] Task configuration if given in ui5.yaml
+ * @returns {Promise<undefined>} Promise resolving with <code>undefined</code> once data has been written
+ */
+module.exports = async function ({workspace, dependencies, options: {projectName, projectNamespace, configuration}}) {
+    ws = workspace;
+    rootDir = "/resources/" + projectNamespace;
+    if (!configuration) {
+        throw "Configuration is missing";
+    }
+    switch (configuration.serviceWorker.strategy) {
+        case swStrategies.offlinePage:
+            await addOfflinePageSw(configuration.serviceWorker);
+            break;
+        case swStrategies.offlineCopy:
+            await addOfflineCopySw();
+            break;
+        case swStrategies.offlineCopyWithBackup:
+            await addOfflineCopyBackupSw(configuration.serviceWorker);
+            break;
+        case swStrategies.preCache:
+            await addCacheFirstSw(configuration.serviceWorker);
+            break;
+        case swStrategies.advancedCache:
+            await addAdvancedCachingSw(configuration.serviceWorker);
+            break;
+        default:
+            throw "Unknown service worker strategy";
+    }
+    await addServiceworkerRegistration();
+    await addManifest(configuration.manifest);
+    return undefined;
+};
+
+/**
+ * Adds the script tag required to register a service worker to the index.html, the added code will register
+ * service-worker.js as the service worker for this website.
+ *
+ * @returns {Promise<undefined>}
+ */
+async function addServiceworkerRegistration() {
+    let fileContent = fs.readFileSync(path.join(__dirname, "../templates/register-sw.js"));
+    await appendToIndexHtmlHead("<script>" + fileContent.toString() + "</script>");
+    return undefined;
```

Comment:
see above - for `async` functions returning no value means a `Promise.resolve` with `undefined`.